src/deepsparse/transformers/server/utils.py

```diff
@@ -0,0 +1,175 @@
+    API level settings for server
+
+    :param host: str representing the host URL for server deployment. Defaults
+        to `0.0.0.0`
+    :param port: int representing the port number to use for server deployment.
+        Defaults to port 5543. Note-> the port must be available for use
+    :param workers: int The number of server processes to spawn. Is set to a
+        constant value of 1 to ensure only one copy of backend deepsparse ENGINE
+        is running at all times.
+    """
+
+    host: str = "0.0.0.0"
+    port: int = 5543
+    workers: int = Field(WORKERS, const=True)
+
+
+def parse_api_settings() -> APIConfig:
+    """
+    Adds an ArgumentParser for DeepSparse Inference Server
+    and parses APIConfig via Command Line
+
+    :return: APIConfig instance for server config
+    """
+    parser = argparse.ArgumentParser("Run DeepSparse Inference Server")
+
+    parser.add_argument(
+        "--host",
+        "-H",
+        type=str,
+        default="0.0.0.0",
+        help="The IP address of the hosted model",
+    )
+    parser.add_argument(
+        "-p",
+        "--port",
+        type=int,
+        default="5543",
+        help="The port that the model is hosted on",
+    )
+
+    parser.add_argument(
+        "-w",
+        "--workers",
+        type=int,
+        default=1,
+        help="The number of workers to use for uvicorn",
+    )
+
+    _args = parser.parse_args()
+    return APIConfig(**vars(_args))
+
+
+# PipelineEngineConfig
+
+DEFAULT_BATCH_SIZE = 1
+DEFAULT_MAX_LENGTH = 128
+DEFAULT_SCHEDULER = "multi"
+PipelineConfigType = TypeVar(
+    "PipelineEngineSettingsType",
+    bound="PipelineEngineConfig",
+)
+
+
+class PipelineEngineConfig(BaseSettings):
+    """
+    Settings for pipeline should be set via .env file
+    using the following steps:
+    ```
+    set -a
+    export <file>.env
+    set +a
+    ```
+
+    :param model_file_or_stub: path to (ONNX) model file or SparseZoo stub
+    :param task: name of the task to define which pipeline to create.
+    :param num_cores: number of CPU cores to run ENGINE with. Default is the
+        maximum available
+    :param batch_size: The batch size to use for pipeline. Defaults to 1.
+        Note: "question-answering" only supports batch_size 1
+    :param max_length: maximum sequence length of model inputs. Default is 128
+    :param scheduler: The scheduler to use for the ENGINE. Can be None, single
+        or multi
+    :param concurrent_engine_requests: Number of concurrent workers to use
+        for sending requests to the Pipeline. Defaults to 3.
+    """
+
+    task: str
+    model_file_or_stub: str = (
+        "zoo:nlp/question_answering/bert-base/"
+        "pytorch/huggingface/squad/"
+        "pruned_quant_3layers-aggressive_89"
+    )
+    num_cores: Optional[int] = None
+    batch_size: Optional[int] = DEFAULT_BATCH_SIZE
+    max_length: int = DEFAULT_MAX_LENGTH
+    scheduler: str = DEFAULT_SCHEDULER
+    concurrent_engine_requests: int = 3
```

Comment:
not sure why I missed this, but we should not be defaulting this to 3. Let's do `num_cores / 2` (default to `None` and set it to `num_cores / 2` if so)

Reply:
Done